### PR TITLE
[consensus] Propagate commit QC

### PR DESCRIPTION
**Pull request by ikabiljo**
Leaving fixing of the bug and fetching appropriate blocks to a separate diff

### Test Plan

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1813)

### CODE REVIEW

File: consensus/consensus-types/src/quorum_cert.rs [hunk 1]
```diff
@@ -126,4 +126,25 @@ impl QuorumCert {
         self.vote_data.verify()?;
         Ok(())
     }
+
+    pub fn create_merged_with_executed_state(
+        &self,
+        executed_ledger_info: LedgerInfoWithSignatures,
+    ) -> anyhow::Result<QuorumCert> {
+        let self_commit_info = self.commit_info();
+        let executed_commit_info = executed_ledger_info.ledger_info().commit_info();
```
Comment (zekun000) [inline consensus/consensus-types/src/quorum_cert.rs:135]: there's a function `match_ordered_only` in block info for this purpose

File: consensus/consensus-types/src/sync_info.rs [hunk 2]
```diff
@@ -47,7 +44,7 @@ impl SyncInfo {
     pub fn new_decoupled(
         highest_quorum_cert: QuorumCert,
         highest_ordered_cert: QuorumCert,
-        highest_ledger_info: Option<LedgerInfoWithSignatures>,
+        highest_commit_cert: Option<QuorumCert>,
```
Comment (zekun000) [inline consensus/consensus-types/src/sync_info.rs:47]: let's make it non-option, i think the option is there to provide backward compatibility

File: consensus/src/block_storage/block_store.rs [hunk 3]
```diff
@@ -237,8 +237,11 @@ impl BlockStore {
     }
 
     /// Commit the given block id with the proof, returns () on success or error
-    pub async fn commit(&self, finality_proof: LedgerInfoWithSignatures) -> anyhow::Result<()> {
-        let block_id_to_commit = finality_proof.ledger_info().consensus_block_id();
+    pub async fn commit(&self, finality_proof: QuorumCert) -> anyhow::Result<()> {
+        let block_id_to_commit = finality_proof
+            .ledger_info()
+            .ledger_info()
```
Comment (zekun000) [inline consensus/src/block_storage/block_store.rs:243]: this is ugly lol, we can do commit_info().id()

**Review by zekun000 — COMMENTED** [review]
overall looks great!

File: consensus/consensus-types/src/sync_info.rs (hunk 2 above)
Comment (ikabiljo) [reply consensus/consensus-types/src/sync_info.rs:47]: we seem to be setting it to None here - https://github.com/aptos-labs/aptos-core/blob/ac9eeaf35d7d18a09afda38f92e1e563f19fa333/consensus/consensus-types/src/sync_info.rs#L55, so I can remove it from this function, but not from SyncInfo struct?

Comment (zekun000) [reply consensus/consensus-types/src/sync_info.rs]: this Option is for highest ordered cert but not commit_cert?

**Review by zekun000 — APPROVED** [review]

**Pull request merged**